### PR TITLE
Add gres support for AMD MI300 GPUs

### PR DESCRIPTION
**Pull request by bwatrous**

### CODE REVIEW

File: templates/slurm.txt [hunk 1]
```diff
@@ -246,7 +246,7 @@ Order = 1
...
...tr><td><p>Slurm provides massive scalability and can easily manage performance requirements for small cluster, large cluster, and supercomputer needs. Slurm outperforms competitive schedulers with compute rates at:</p></td></tr><tr><td><ul><li>100K+ nodes/GPU</li><li>17M+ jobs per day</li><li>120M+ jobs per week</li></ul></td></tr><tr><td><br><p>Slurm’s plug-in based architecture enables optimization and control in scheduling operations to meet organizational priorities. With first class resource management for GPUs, Slurm allows users to request GPU resources alongside CPUs. This flexibility ensures that jobs are executed quickly and efficiently, while maximizing resource utilization.</p><br></td></tr><tr><td><p>Other Slurm features include:</p></td></tr><td><ul><li>NVIDIA and AMD GPU support for AI, LLM, and ML environments</li><li>Advanced scheduling policies</li><li>Unique HPC, HTC, AI/ML workload expertise</li><li>Cloud bursting capabilities</li><li>Power saving capabilities, accounting, and reporting</li><li>Provided REST API daemon</li><li>Native support of containers</li><li>Tailored Slurm consulting and training available through SchedMD</li></ul></td></table></details>'''
```
Comment (aditigaur4) [inline templates/slurm.txt:249]: Is this change intentional? hard to understand the diff here. what are we trying to change?

File: slurm/src/slurmcc/cli.py [hunk 2]
```diff
@@ -731,13 +731,23 @@ def _generate_gres_conf(partitions: List[partitionlib.Partition], writer: TextIO
...
             if partition.gpu_count:
+                # TODO: azure sku should eventually indicate the GPU type
+                try:
+                    has_amd_gpu = partition.gpu_family.lower() == "amd"
+                except AttributeError:   
+                    has_amd_gpu = "mi300" in partition.machine_type.lower()
+
                 if partition.gpu_count > 1:
-                    nvidia_devices = "/dev/nvidia[0-{}]".format(partition.gpu_count - 1)
+                    nvidia_devices = "/dev/nvidia[0-{}]".format(partition.gpu_count - 1)
```
Comment (aditigaur4) [inline slurm/src/slurmcc/cli.py:741]: Is it possible to move this into a "generate_nvidia_gpu_device()" function of some sort and then  just give it number of devices to generate? same for amd devices.

File: templates/slurm.txt (hunk 1 above)
Comment (bwatrous) [reply templates/slurm.txt:249]: Yeah - basically just changed from "NVidia GH200 support" to "NVIDIA and AMD GPU support"

**Review by aditigaur4 — APPROVED** [review]
